### ISSUE TITLE
Write the ah_http library

### ISSUE THREAD

**Issue by emanuelpalm** [eclipse-arrowhead/core-c#9]
The library shall support HTTP/1.0 and HTTP/1.1, as well as be suitable for use by embedded systems. This means that it is acceptable for the library user to decide how many headers and what headers can be in an expected message beforehand, as well as how much memory the header of an HTTP message may occupy.

The library must support

- [x] using both plain TCP and TLS as transport,
- [x] HTTP version negotiation,
- [x] all HTTP methods,
- [x] arbitrary HTTP headers, 
- [x] chunked encoding, 
- [x] treat the `host` header correctly,
- ~give the `accept` and `content-length` headers special treatment, such that interpreting them becomes more convenient for the library user~.

In other words, the library does not implement all of HTTP/1.1, but enough for the library to be useful for using HTTP for RESTful Arrowhead services.

The library shall do nothing to help interpret the contents of any given message, nor help with decoding form data. It shall not compress or decompress message bodies automatically. Compression is useful primarily for larger unencrypted payloads, which we deem to be atypical for the use cases we have in mind (in which rather smaller encrypted payloads are sent).
